### PR TITLE
Recover from corrupted cache service SQLite files

### PR DESCRIPTION
**Pull request by kraih**
I've tried quite a few variants of this on the staging machines, and this is the one that performed best for recovering from various cases of corrupted SQLite files on service restart.

Possible addition: The purge code could also make a copy of the corrupted database to `/tmp/cache.sqlite` (maybe there's a better location?) for later investigation.

Progress: https://progress.opensuse.org/issues/67000

### CODE REVIEW

File: t/25-cache.t [hunk 1]
```diff
@@ -51,7 +51,7 @@ use POSIX '_exit';
 use Mojo::IOLoop::ReadWriteProcess 'process';
 use Mojo::IOLoop::ReadWriteProcess::Session 'session';
 use OpenQA::Test::Utils qw(fake_asset_server wait_for_or_bail_out);
-use OpenQA::Test::TimeLimit '20';
+use OpenQA::Test::TimeLimit '60';
```
Comment (okurz) [inline t/25-cache.t:54]: Hm, so does it take really three times as long as in before? We should care to have speedy tests. Do you think you can maybe take some shortcuts in tests or mock to speed up?

Comment (Martchus) [reply t/25-cache.t:54]: > so does it take really three times as long as in before?

I've already noticed that the runtime of some tests is too low when running them locally. I suppose it only works within the CI due to the scale factor. Maybe that's another one of these cases.

Comment (kraih) [reply t/25-cache.t:54]: No, i was just being generous because it took longer than 20 seconds in my VM to run the test before i even added anything new.

Comment (kraih) [reply t/25-cache.t:54]: But good to know i'm not the only one running into local tests being slower than on CI, even on a fast machine.

Comment (Martchus) [reply t/25-cache.t:54]: > running into local tests being slower than on CI

As mentioned, the CI has the scale factor. So the limits are just higher within the CI; it is most likely not faster within the CI.

Comment (kraih) [reply t/25-cache.t]: Ah, i missed that there's a scale factor. I'll try reducing the value a bit then.

**Review by Martchus — APPROVED** [review]

File: t/25-cache.t (hunk 1 above)
Comment (Martchus) [reply t/25-cache.t]: Actually, set it to the local execution time plus a small margin. The local execution should not rely on any scaling factor (unless you want to gather coverage data locally).

**Review by okurz — APPROVED** [review]

File: t/25-cache.t (hunk 1 above)
Comment (okurz) [reply t/25-cache.t]: We even have the test time limits documented but the scaling aspect was not covered yet so I did that now in https://github.com/os-autoinst/openQA/pull/3499

**Pull request merged**